docusaurus/website/docusaurus.config.dev.ts

```diff
@@ -32,7 +31,7 @@ const config = {
         },
         theme: presetsTheme,
         blog: false,
-      },
+      } satisfies Preset.Options,
```

Comment:
Changes in docusaurus 3.6.3 package cause typescript types referencing `@theme` to be lost. Previously they were set within the docusaurus-tsconfig like so:

```
    "types": [
      "node",
      "@docusaurus/module-type-aliases",
      "@docusaurus/theme-classic"
    ],
```
This was removed in latest release because the docusaurus team don't want to overwrite the default `types` which prevents automatically loading all `@types` packages within users sites.

There are multiple ways to solve it but given we half use types via `@ts-check` it seemed like an improvement to just switch to config ts files and import and use the types here.